### PR TITLE
insurance plan controller and tests

### PR DESCRIPTION
**Pull request by eclendenning-va**

### CODE REVIEW

**Review by jabando-va — APPROVED** [review]

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanWitnessProtectionAgent.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+/** Creates protected references for the InsurancePlan resource. */
+@Component
+@RequiredArgsConstructor(onConstructor_ = {@Autowired, @NonNull})
+public class R4InsurancePlanWitnessProtectionAgent
+    implements WitnessProtectionAgent<InsurancePlan> {
+  private final ProtectedReferenceFactory protectedReferenceFactory;
+
+  @Override
+  public Stream<ProtectedReference> referencesOf(InsurancePlan resource) {
+    return Stream.of(protectedReferenceFactory.forResource(resource, resource::id));
```
Comment (bschofield-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanWitnessProtectionAgent.java:22]: Does this protect https://github.com/department-of-veterans-affairs/health-apis-vista-fhir-query/blob/master/vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanTransformer.java#L159

Comment (eclendenning-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanWitnessProtectionAgent.java:22]: No, i'm working on that now. I wanted to get a version without full witness protection up first. I'll try and add it before this merges.

Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanWitnessProtectionAgent.java:22]: I can get on board with two PRs.... if there is more coming, this looks good enough to merge now... your call.

Comment (eclendenning-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/insuranceplan/R4InsurancePlanWitnessProtectionAgent.java]: Should be good to go now 👍

**Review by ilaflamme-va — APPROVED** [review]

**Review by bschofield-va — APPROVED** [review]

**Pull request merged**